jacodb-analysis/src/main/kotlin/org/jacodb/analysis/codegen/Generator.kt

```diff
@@ -164,31 +165,42 @@ fun main(args: Array<String>) {
     // 3. process exit code - ok
     // 4. if not ok - stdout
     // 5. java_home > 1.8
-    System.getenv()
-
-    val java_home = System.getProperty("java.version")
-    if (java_home != null && java_home >= "1.8") {
+    val javaHome1 = System.getProperty("java.version")
+    if (javaHome1 != null && javaHome1 >= "1.8") {
         val isWindows = System.getProperty("os.name").toLowerCase().startsWith("windows")
         val processBuilder = ProcessBuilder()
-        processBuilder.directory(
-            File("generated\\UtBotTemplateForIfdsSyntheticTests".replace('\\', File.separatorChar))
-        )
-        var result = 0
+        val workingDir = File("generated\\UtBotTemplateForIfdsSyntheticTests".replace('\\', File.separatorChar))
+        processBuilder.directory(workingDir)
+        val errorFile = File(workingDir.absolutePath + File.separatorChar + "gradlewErrors")
+        processBuilder.redirectError(errorFile)
         if (isWindows) {
             processBuilder.command("./gradlew", "assemble")
         } else {
-//            listOf(
-//                listOf("./gradlew", "assemble"),
-//                listOf("chmod", "+x", "gradlew")
-//            ).forEach { cmd -> processBuilder.command(cmd) }
-            processBuilder.command("chmod +x gradlew; ./gradlew assemble")
+            val chmodOutput = chmodGradlew(workingDir)
+            if (chmodOutput.isNotEmpty()) {
+                logger.info("problems with chmod:$chmodOutput")
+            } else {
+                processBuilder.command("./gradlew", "assemble")
+            }
         }
-        result += processBuilder.start().waitFor()
-        if (result != 0) {
-            logger.info("gradle assembling ended with output value: $result")
-//            println("problems with gradle")
+        processBuilder.start()
+        val result = Files.readAllLines(errorFile.toPath())
```

Comment:
this does not work as you dont wait until process ends

Reply:
And so your file will always be empty

Reply:
done